dask/dataframe/shuffle.py

```diff
@@ -71,7 +71,8 @@ def set_index(df, index, npartitions=None, shuffle=None, compute=False,
                 divisions = [divisions[i] for i in indexes]
 
         if (mins == sorted(mins) and maxes == sorted(maxes) and
```

Comment:
I'm starting to think this isn't the correct fix. AFAIU, this `if` block is ensuring that `index` is already sorted, skipping an expensive sort, correct?

I'm still trying to grok the code, but `mins` and `maxes` are just the partition points. How do we know that it's sorted within each of those?

I'm digging into `repartition_quantiles` now... I'm worried that it's an issue with us using approximate quantiles, when in this case it seems like it has to be exact.